ocaml/myocamlbuild.ml

```diff
@@ -69,23 +69,27 @@ let list_dependencies () =
   String.concat "\n" lines
 
 
-let major_minor_patch () =
-  let tag_version = run_cmd "git describe --tags --exact-match --dirty"
-  and branch_version = run_cmd "git describe --all" in
-  let (major, minor, patch) =
+let major_minor_patch_modifier () =
+  let tag_version = run_cmd "git describe --tags --dirty" in
+  let tv =
+    try tag_version ()
+    with _ -> ""
+  in
+  let (major, minor, patch, modif) =
     try
-      Scanf.sscanf (tag_version ()) "%i.%i.%i" (fun ma mi p -> (ma,mi,p))
+      Scanf.sscanf tv "%i.%i.%i-%s" (fun ma mi p dev -> (ma,mi,p, Some dev))
     with _ ->
-      let bv = branch_version () in
```

Comment:
this change means branch name will no longer be in the version output when not building exactly a tagged version? - which I did consider something useful to be honest